tests/TestContextTests.elm

```diff
@@ -56,4 +59,68 @@ all =
+                TestContext.createWithFlags
+                    { init = \flags -> "<INIT:" ++ flags ++ ">"
+                    , update = testUpdate
+                    , view = testView
+                    }
+                    "flags"
+                    |> TestContext.expectModel (Expect.equal "<INIT:flags>")
+                    |> TestContext.done
+        , test "can create with navigation" <|
+            \() ->
+                TestContext.createWithNavigation
+                    .pathname
+                    { init = \location -> "<INIT:" ++ location.pathname ++ ">"
+                    , update = testUpdate
+                    , view = testView
+                    }
+                    "https://example.com/path"
+                    |> TestContext.expectModel (Expect.equal "<INIT:/path>")
+                    |> TestContext.done
+        , test "can simulate a route change" <|
+            \() ->
+                TestContext.createWithNavigation
+                    .pathname
+                    { init = \location -> "<INIT:" ++ location.pathname ++ ">"
+                    , update = testUpdate
+                    , view = testView
+                    }
+                    "https://example.com/path"
+                    |> TestContext.routeChange "https://example.com/new"
+                    |> TestContext.expectModel (Expect.equal "<INIT:/path>;/new")
+                    |> TestContext.done
+        , test "can create with navigation and flags" <|
+            \() ->
+                TestContext.createWithNavigationAndFlags
+                    .pathname
+                    { init = \flags location -> "<INIT:" ++ location.pathname ++ ":" ++ flags ++ ">"
+                    , update = testUpdate
+                    , view = testView
+                    }
+                    "https://example.com/path"
+                    "flags"
+                    |> TestContext.expectModel (Expect.equal "<INIT:/path:flags>")
+                    |> TestContext.done
+        , test "can assert on the view" <|
+            \() ->
+                testContext
+                    |> TestContext.shouldHaveView
+                        (Query.find [ Selector.tag "span" ] >> Query.has [ Selector.text "<INIT>" ])
+                    |> TestContext.done
+        , test "can create with navigation and JSON string flags" <|
+            \() ->
+                TestContext.createWithNavigationAndJsonStringFlags
+                    (Json.Decode.field "x" Json.Decode.string)
+                    .pathname
+                    { init = \flags location -> "<INIT:" ++ location.pathname ++ ":" ++ flags ++ ">"
+                    , update = testUpdate
+                    , view = testView
+                    }
+                    "https://example.com/path"
+                    """{"x": "fromJson"}"""
+                    |> TestContext.expectModel (Expect.equal "<INIT:/path:fromJson>")
+                    |> TestContext.done
```

Comment:
🎉 Woohoo, looks good! Do you plan on testing any of the error cases? (I'm okay with no tests, just wondering!)

Reply:
Maybe eventually when it's more possible to control what the error messages look like.  Basically all the tests failed before they were implemented, so I felt comfortable with that.